src/Modules/DashboardModule/components/Dashboard/Dashboard.tsx

```diff
@@ -1,5 +1,209 @@
-import React from "react";
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { Bar, Doughnut } from "react-chartjs-2";
+import { useUser } from "../../../../Context/AuthContext";
+import {
+  baseUrl,
+  getRequestHeaders,
+  handleApiError,
+} from "../../../../utils/Utils";
+import Header from "../../../SharedModule/components/Header";
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  BarElement,
+  DoughnutController,
+  ArcElement,
+  Title,
+  Tooltip,
+  Legend,
+} from "chart.js";
+
+import "./Dashboard.Moule.css";
+
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  BarElement,
+  DoughnutController,
+  ArcElement,
+  Title,
+  Tooltip,
+  Legend
+);
 
 export default function Dashboard() {
-  return <>Dashboard</>;
+  const { userRole } = useUser();
+  const [todoCount, settodoCount] = useState<number>(0);
+  const [progressCount, setprogressCount] = useState<number>(0);
+  const [doneCount, setdoneCount] = useState<number>(0);
+  const [activatedEmployeeCount, setactivatedEmployeeCount] =
+    useState<number>(0);
+  const [deactivatedEmployeeCount, setdeactivatedEmployeeCount] =
+    useState<number>(0);
+
+  const getTasksCount = async () => {
+    try {
+      const response = await axios.get(`${baseUrl}/Task/count`, {
+        headers: getRequestHeaders(),
+      });
+
+      settodoCount(response.data.done);
+      setprogressCount(response.data.inProgress);
+      setdoneCount(response.data.toDo);
+    } catch (error) {
+      handleApiError(error);
+    }
+  };
+
+  const getUserCount = async () => {
+    try {
+      const response = await axios.get(`${baseUrl}/Users/count`, {
+        headers: getRequestHeaders(),
+      });
+      setactivatedEmployeeCount(response.data.activatedEmployeeCount);
+      setdeactivatedEmployeeCount(response.data.deactivatedEmployeeCount);
+    } catch (error) {
+      handleApiError(error);
+    }
+  };
+
+  useEffect(() => {
+    getTasksCount();
+    getUserCount();
+  }, []);
```

Comment:
It is recommended by React to include anything declared outside of the effect inside the dependency array.  So here we should put the function inside the dependency array, but we will encounter another issue, the function is created in every rendered and this will cause the effect to run in every render and might cause an infinite loop.
We can solve this by two ways:
1- by using `useCallback` hook to preserve the reference of the function between renders.
2- make another state variable responsible for re-fethcing the data as we did yesterday and declare the function inside the effect.
![image](https://github.com/RanaFathy123/pms-project/assets/64327685/800a1af2-2293-43b9-91f2-3964b30a201d)
 
You can read more about this here in the React docs:
https://react.dev/reference/react/useEffect#specifying-reactive-dependencies
https://react.dev/reference/react/useEffect#removing-unnecessary-function-dependencies.

Also, we can make one effect for each concern (one for the projects and one for the users). This will prevent calling APIs unnecessarily.